src/map/battle.cpp

```diff
@@ -689,6 +689,12 @@ int battle_calc_cardfix(int attack_type, struct block_list *src, struct block_li
 	t_race2 = status_get_race2(target);
 	s_defele = (tsd) ? (enum e_element)status_get_element(src) : ELE_NONE;
 
+	// When the attacker is a monster, then all bonuses on BF_WEAPON will work and no bonuses on BF_MAGIC
+	if (src->type == BL_MOB) {
```

Comment:
Since this has been wrong forever: Shouldnt we have a config for it?

Reply:
Yeah, the reason I didn't finish this PR is that I was started to think about that too.